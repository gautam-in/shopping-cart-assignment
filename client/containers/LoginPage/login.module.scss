@import "../../styles/colors.scss";

.container {
    display: flex;
    flex-direction: column;
    @media screen and (min-width: 720px) {
        flex-direction: row;
        width: 80%;
        margin: auto;
        & > * {
            flex: 1;
        }
    }
}

.loginIntro {
    text-align: center;
    font-weight: 600;
    margin-bottom: 40px;
    @media screen and (min-width: 720px) {
        text-align: left;
        & > h1 {
            font-size: 30px;
            margin-bottom: 0;
        }
    }
}

.loginForm {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.loginButton {
    background-color: $button-primary;
    border: none;
    color: white;
    padding: 10px;
    width: 90%;
}

.input {
    border: none;
    border-bottom: inset $input-border 2px;
    width: 100%;
    font-size: 16px;

    &:focus {
        outline: none;
        border-bottom: inset $input-border-active 2px;
    }
}

.inputLabel {
    width: 90%;
    margin: 20px 0;
    position: relative;
}

.placeholder {
    position: absolute;
    top: 0;
    left: 0;
    transition: top 0.3s ease, color 0.3s ease;
}

input:valid + .placeholder,
input:focus + .placeholder {
    top: -24px;
    color: $input-border-active;
}
