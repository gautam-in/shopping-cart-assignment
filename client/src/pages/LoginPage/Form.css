.container {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 20px;
}

input {
  border: none;
  outline: 1px solid transparent;
}

.container form {
  display: flex;
  justify-content: space-evenly;
  align-items: flex-start;
  width: 100%;
  padding: 20px;
}

.container form .left-side {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 50%;
}

.container form .left-side .heading {
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.container form .right-side {
  display: flex;
  flex-direction: column;
  width: 50%;
}

.container form .right-side .field {
  position: relative;
  display: flex;
  flex-direction: column;
  padding: 10px 20px;
  transition: all 200ms ease-in;
}

.container form .right-side .field input {
  border-bottom: 2px solid #d6d3d3;
  width: 60%;
  padding: 10px 0px;
}

.container form .right-side .field input:focus {
  border-bottom: 2px solid var(--input-field-color);
}

.container form .right-side .field label {
  position: absolute;
  top: -2px;
  letter-spacing: 1px;
  font-size: 0.9rem;
  color: #888888;
  font-weight: bold;
  transition: 0.5s;
}

.container form .right-side .field input:focus ~ label {
  color: var(--input-field-color);
  font-size: 0.8rem;
}

.container form .right-side .field p {
  margin: 10px 0;
  font-weight: bold;
  font-size: 0.9rem;
  color: red;
}

.container form .right-side .button {
  padding: 10px;
  width: 60%;
  margin: 10px;
  font-size: 1.2rem;
}

@media only screen and (max-width: 1024px) {
  .container form {
    min-width: 100vw;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .container form .left-side {
    width: 100%;
    margin-bottom: 5rem;
  }

  .container form .right-side {
    width: 100%;
    justify-content: center;
    align-items: center;
  }

  .container form .right-side .field {
    width: 100%;
    justify-content: center;
    align-items: center;
    flex-direction: column-reverse;
  }

  .container form .right-side .field label {
    position: relative;
  }
}
