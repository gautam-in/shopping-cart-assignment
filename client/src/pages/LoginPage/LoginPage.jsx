import React, { useState, useEffect, useCallback } from 'react';
import styled from 'styled-components';
import { useDispatch } from 'react-redux';

import Button from '../../components/Button/Button';
import FormInput from '../../components/FormInput/FormInput';

import { login } from './../../redux/User/actions';
import { validate } from './../../utils/helpers';

const LoginContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  column-gap: 50px;
  padding-top: 10%;

  // Tablets
  @media only screen 
  and (min-width : 768px) 
  and (max-width : 1024px) {
    padding: 0 40px;
    padding-top: 10%;
  }

  // Large screen mobiles
  @media only screen 
  and (min-width : 321px) 
  and (max-width : 480px) {
    flex-direction: column;
    padding: 0 40px;
    padding-top: 10%;
  }
`;

const Title = styled.div`
  display: flex;
  flex-direction: column;
  row-gap: 20px;
`;

const Form = styled.form`
  width: 35%;
  input[type=text] {
    -webkit-text-security: disc;
    text-security: disc;
  }

  // Large screen mobiles
  @media only screen 
  and (min-width : 321px) 
  and (max-width : 480px) {
    width: 100%;
  }  
`;

const CustomButton = styled(Button)`
  width: 100%;
  &:disabled {
    opacity: 0.7;
    cursor: not-allowed;
  }
`;

const LoginPage = () => {
  const dispatch = useDispatch();
  const [userData, setUserData] = useState({
    email: '',
    pwd: ''
  });

  const [errors, setErrors] = useState({});
  const [touched, setTouched] = useState({});
  const [formValid, setFormValid] = useState(false);

  const checkFormValidity = useCallback(() => {
    const haveAnyErrors = !!Object.values(errors).find(err => err.hasError);
    const touchedFields = Object.values(touched).filter(item => item);
    const haveTouchedAllFields = Object.values(userData).length === touchedFields.length;
    const isValid = !haveAnyErrors && haveTouchedAllFields;
    setFormValid(isValid);
  }, [errors, touched, userData]);

  useEffect(() => {
    checkFormValidity();
  }, [userData, checkFormValidity]);

  const handleChange = ev => {
    const { name, value } = ev.target;
    const { hasError, errMessage } = validate(ev.target);
    if(!touched[name]) {
      setTouched((prevState) => {
        return {
          ...prevState,
          [name]: true
        }
      });
    }
    setErrors((prevState) => {
      return {
        ...prevState,
        [name]: { hasError, errMessage }
      }
    });
    setUserData((prevState) => {
      return {
        ...prevState,
        [name]: value
      }
    });
  }

  const handleSubmit = ev => {
    ev.preventDefault();
    dispatch(login(userData));
    setUserData({
      email: '',
      pwd: ''
    });
    setErrors({});
    setTouched({});
    setFormValid(false);
  };

  const { email, pwd } = userData;
  return (
    <LoginContainer>
      <Title>
        <h2>Login</h2>
        <p>Get access to your Orders, Wishlist and Recommendations</p>
      </Title>
      <Form onSubmit={handleSubmit}>
        <FormInput
          type="email"
          name="email"
          label="Email"
          value={email}
          required
          onChange={handleChange}
          errors={errors?.email} />
        <FormInput
          type="text"
          name="pwd"
          label="Password"
          value={pwd}
          required
          onChange={handleChange}
          errors={errors?.pwd} />
        <CustomButton
          type="submit"
          disabled={!formValid}>
            Login
        </CustomButton>
      </Form>
    </LoginContainer>
  );
}

export default LoginPage;