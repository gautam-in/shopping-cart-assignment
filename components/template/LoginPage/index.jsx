import React, {Component} from "react";
import PageLayout from "../../organism/PageLayout";
import LoginForm from "../../organism/LoginForm";
import {Row, Col} from 'reactstrap';
class LoginPage extends Component {
    render(){
        return(
            <PageLayout>
                <Row>
                    <Col md={6} className="pt-3 pl-5">
                        <h2>Login</h2>
                        <p>Get access to your Orders, Wishlist and recommendations</p>
                    </Col>
                    <Col md={6}>
                        <LoginForm/>
                    </Col>
                </Row>                
            </PageLayout>    
            )
    }
}

export default LoginPage;