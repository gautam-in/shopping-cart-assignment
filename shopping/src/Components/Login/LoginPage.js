import React from 'react';
import ReactDOM from 'react-dom'; 
import { Container, Row ,Col,} from 'react-bootstrap';
import './login.scss'
const LoginPage = () => {
            return (
              <div  className="container" id="login-page">
                  <div className="row  mt-3" >
                    <div className="col  m6 s12 ">
                      nkjn
                    </div>
                    <form className="col  m6 s12">
                      <div className="row m-0">
                        <div className="input-field col s12 m-0">
                          <input  id="email" type="text" className="validate"/>
                          <label for="email">Email</label>
                        </div>
                        </div>
                        <div className="row m-0">
                        <div className="input-field col s12 m-0">
                          <input id="password" type="password" className="validate"/>
                          <label for="password">Password</label>
                        </div>
                      </div>
                    
                    </form>
                    </div>
          </div>
          );
  };

export default LoginPage;