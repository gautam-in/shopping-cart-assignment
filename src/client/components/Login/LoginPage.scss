.section-login {
    display: flex;
    justify-content: center;
    .login-info {
        line-height: 2.25rem;
        flex: 0 0 30%;
        p {
            font-size: 70%;
            font-weight: 600;
        }
    }
    .login-action {
        flex: 0 0 25%;
        .login-form {
            .login-inputs {
                width: 100%;
            }
        }
        .field-row {
            margin-bottom: 1rem;
            width: 12rem;
        }

        .btn-login,
        .btn-login:active,
        .btn-login:focus {
            font-size: 90%;
            display: block;
            line-height:2rem;
            margin: 2rem 0 0 0;
            width: 100%;
            outline: none;
            border: none;
            cursor: pointer;
        }
    }
}
