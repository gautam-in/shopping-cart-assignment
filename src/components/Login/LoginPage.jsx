import React from 'react';
import { TextField } from '@material-ui/core';
import './LoginPage.scss';

const LoginPage = (props) => {
  const { fields, formActions, onSubmit, onInputChange } = props;
  return (
    <section className="section-login">
      <div className="login-info">
        <h2>Login</h2>
        <p>Get access to your Orders, Wishlist and Recommendations</p>
      </div>
      <div className="login-action">
        <form method="post" action="#" className="login-form" autoComplete="off" onSubmit={(e) => onSubmit(e, {...fields})}>
          <div className="field-row">
            <TextField
              name="email"
              onChange={(e) => onInputChange(e, fields.email)}
              value={fields.email.value}
              className="login-inputs"
              id="email"
              type="email"
              label="Email"
            />
            {fields.email.error && <span className="error-help">{fields.email.error}</span>}
          </div>
          <div className="field-row">
            <TextField
              name="password"
              onChange={(e) => onInputChange(e, fields.password)}
              value={fields.password.value}
              className="login-inputs"
              id="password"
              type="password"
              label="Password"
            />
            {fields.password.error && <span className="error-help">{fields.password.error}</span>}
          </div>
          <button
            type="submit"
            className="btn btn-login"
            disabled={formActions.isSubmitting || formActions.hasErrors}
          >
            Login
          </button>
        </form>
      </div>
    </section>
  )
};

export default LoginPage;
