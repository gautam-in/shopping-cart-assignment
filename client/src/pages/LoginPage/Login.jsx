import React from "react";
import { useForm } from "react-hook-form";
import { useNavigate } from "react-router";
import Button from "../../components/Button/Button";
import "./Form.css";

function Login() {
  const navigate = useNavigate();
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  const onSubmit = (data) => {
    navigate("/");
  };

  return (
    <div className="container">
      <form action="post" onSubmit={handleSubmit(onSubmit)}>
        <div className="left-side">
          <div className="heading">
            <h1>Login</h1>
            <p>Get access to your Orders, Wishlist and Recommendations</p>
          </div>
        </div>

        <div className="right-side">
          <div className="field">
            <input
              {...register("email", { required: "Email is required" })}
              type="email"
              id="email"
              required
            />
            <label htmlFor="email">Email</label>
            <p>{errors.email?.message}</p>
          </div>
          <div className="field">
            <input
              {...register("password", { required: "Password is required" })}
              type="password"
              id="password"
              autoComplete="off"
              required
            />
            <label htmlFor="password">Password</label>
            <p>{errors.password?.message}</p>
          </div>
          <Button className="button" type="submit">
            Login
          </Button>
        </div>
      </form>
    </div>
  );
}

export default Login;
