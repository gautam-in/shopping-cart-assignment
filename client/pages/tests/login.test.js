import React from "react";
import { render } from "@testing-library/react";
import LoginPage from "../login.page";

describe("Login", () => {
  it("should render the component correctly", () => {
    const LoginPageComponent = render(<LoginPage />);
    expect(LoginPageComponent).toMatchSnapshot();
  });
});
