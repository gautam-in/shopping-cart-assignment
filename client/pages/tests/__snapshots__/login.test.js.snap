// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="SectionWrapperstyles__SectionWrapper-sc-1qizg9n-0 gZaGKd"
      >
        <div
          class="LoginFormstyles__LoginFormWrapper-sc-wnnuzy-0 hdmsQj"
        >
          <section
            class="LoginFormstyles__LoginFormHeading-sc-wnnuzy-1 hcmeBz"
          >
            <h1>
              Login
            </h1>
            <p>
              Get access to your Orders, Wishlist and Recommendations.
            </p>
          </section>
          <form
            class="LoginFormstyles__Form-sc-wnnuzy-2"
          >
            <div
              class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
            >
              <input
                class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
                id="email"
                name="email"
                title="Email"
                type="email"
              />
              <label
                class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
                for="email"
              >
                Email
              </label>
            </div>
            <div
              class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
            >
              <input
                class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
                id="password"
                name="password"
                title="Password"
                type="password"
              />
              <label
                class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
                for="password"
              >
                Password
              </label>
            </div>
            <button
              aria-disabled="true"
              class="Buttonstyles__SubmitButton-sc-5pqk3u-0 exksWu"
              disabled=""
              id="submit"
              name="Login"
              title="Login"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="SectionWrapperstyles__SectionWrapper-sc-1qizg9n-0 gZaGKd"
    >
      <div
        class="LoginFormstyles__LoginFormWrapper-sc-wnnuzy-0 hdmsQj"
      >
        <section
          class="LoginFormstyles__LoginFormHeading-sc-wnnuzy-1 hcmeBz"
        >
          <h1>
            Login
          </h1>
          <p>
            Get access to your Orders, Wishlist and Recommendations.
          </p>
        </section>
        <form
          class="LoginFormstyles__Form-sc-wnnuzy-2"
        >
          <div
            class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
          >
            <input
              class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
              id="email"
              name="email"
              title="Email"
              type="email"
            />
            <label
              class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
              for="email"
            >
              Email
            </label>
          </div>
          <div
            class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
          >
            <input
              class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
              id="password"
              name="password"
              title="Password"
              type="password"
            />
            <label
              class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
              for="password"
            >
              Password
            </label>
          </div>
          <button
            aria-disabled="true"
            class="Buttonstyles__SubmitButton-sc-5pqk3u-0 exksWu"
            disabled=""
            id="submit"
            name="Login"
            title="Login"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
